src/libstore/windows/pathlocks.cc

```diff
@@ -1,16 +1,151 @@
 #include "logging.hh"
 #include "pathlocks.hh"
+#include "signals.hh"
+#include "util.hh"
+#include <errhandlingapi.h>
+#include <fileapi.h>
+#include <windows.h>
+#include "windows-error.hh"
 
 namespace nix {
 
-bool PathLocks::lockPaths(const PathSet & _paths, const std::string & waitMsg, bool wait)
+void deleteLockFile(const Path & path, Descriptor desc) {
+
+  int exit = DeleteFileA(path.c_str());
+  if (exit == 0)
+    warn("%s: &s", path, std::to_string(GetLastError()));
+}
+
+void PathLocks::unlock()
+{
+    for (auto & i : fds) {
+        if (deletePaths) deleteLockFile(i.second, i.first);
+
+        if (CloseHandle(i.first) == -1)
+            printError(
+                "error (ignored): cannot close lock file on '%1%'",
+                i.second);
+
+        debug("lock released on '%1%'", i.second);
+    }
+
+    fds.clear();
+}
+
+
+AutoCloseFD openLockFile(const Path & path, bool create)
+{
+  AutoCloseFD desc = CreateFileA(path.c_str(),
```

Comment:
Nitpick but this function (and some others) aren't indented correctly (i.e. 4 spaces).